modules/nlu-testing/src/tools/data_loader.ts

```diff
@@ -0,0 +1,135 @@
+import * as sdk from 'botpress/sdk'
+import crypto from 'crypto'
+import _ from 'lodash'
+
+import { BotState, Data, RawData, Test } from '../backend/typings'
+
+async function splitTrainToTrainAndTest(state: BotState) {
+  // Backup the real intent folder
+  if (!(await state.ghost.fileExists('./', 'raw_intents'))) {
+    const intentsFiles = await state.ghost.directoryListing('./intents', '*.json')
+    for (const file of intentsFiles) {
+      const intentData = await state.ghost.readFileAsObject<RawData>('./intents', file)
+      await state.ghost.upsertFile('./raw_intents', file, JSON.stringify(intentData, undefined, 2))
+    }
+  }
+
+  const intentsFiles = await state.ghost.directoryListing('./raw_intents', '*.json')
+  let tests: Test[] = []
+  for (const file of intentsFiles) {
+    const intentDatas = await state.ghost.readFileAsObject<RawData>('./raw_intents', file)
+    const languages = Object.keys(intentDatas.utterances)
+
+    for (const lang of languages) {
+      const test_utts = _.sampleSize(intentDatas.utterances[lang], _.floor(intentDatas.utterances[lang].length / 4))
+      const train_utts = intentDatas.utterances[lang].filter(s => !test_utts.includes(s))
+      intentDatas.utterances[lang] = train_utts
+
+      tests = tests.concat(
+        test_utts.map(s => {
+          return {
+            id: crypto
+              .createHash('md5')
+              .update(s)
+              .digest('hex'),
+            conditions: [
+              ['context', 'is', intentDatas.contexts[0]],
+              ['intent', 'is', intentDatas.name]
+            ],
+            utterance: s,
+            context: intentDatas.contexts[0]
+          } as Test
+        })
+      )
+    }
+
+    await state.ghost.upsertFile('./intents', file, JSON.stringify(intentDatas, undefined, 2))
+  }
+  await state.ghost.upsertFile('./', 'nlu-tests.json', JSON.stringify(tests, undefined, 2))
+}
+
+export async function getTrainTestDatas(state: BotState, logger: sdk.Logger, engine) {
```

Comment:
Super weird that engine has no type and is actually the static class instead of an instance. I personnaly would only pass `bp: typesof sdk` as a dependency

Reply:
Changed : Now the function get only bp and state as parameter.